pcdsdevices/mv_interface.py

```diff
@@ -120,3 +134,352 @@ def umvr(self, delta, timeout=None):
+        `FltMvInterface`.
+
+    Attributes
+    ----------
+    positions: ``SimpleNamespace``
+        A namespace that contains all of the active presets as `PresetPosition`
+        objects.
+    """
+    _paths = {}
+
+    def __init__(self, device):
+        self._device = device
+        self._methods = []
+        self.name = device.name + '_presets'
+        self.sync()
+
+    def _path(self, preset_type):
+        """
+        Utility function go get the proper ``Path`` object that points to the
+        presets file.
+        """
+        path = self._paths[preset_type] / (self._device.name + '.yml')
+        logger.debug('select presets path %s', path)
+        return path
+
+    def _read(self, preset_type):
+        """
+        Utility function to get a particular preset's datum dictionary.
+        """
+        logger.debug('read presets for %s', self._device.name)
+        with self._path(preset_type).open('r') as f:
+            return yaml.load(f)
+
+    def _write(self, preset_type, data):
+        """
+        Utility function to overwrite a particular preset's datum dictionary.
+        """
+        logger.debug('write presets for %s', self._device.name)
+        with self._path(preset_type).open('w') as f:
+            yaml.dump(data, f, default_flow_style=False)
+
+    def _update(self, preset_type, name, value=None, comment=None,
+                active=True):
+        """
+        Utility function to read the existing preset's datum, update the value
+        the comment, and the active state, and then write the datum back to the
+        file, updating the last updated time and history accordingly.
+        """
+        logger.debug(('call %s presets._update(%s, %s, value=%s, comment=%s, '
+                      'active=%s)'), self._device.name, preset_type, name,
+                     value, comment, active)
+        try:
+            data = self._read(preset_type)
+        except FileNotFoundError:
+            data = {}
+        if name not in data:
+            data[name] = {}
+        if value is not None:
+            ts = time.strftime('%d %b %Y %H:%M:%S')
+            data[name]['value'] = value
+            history = data[name].get('history', {})
+            history[ts] = value
+            data[name]['history'] = history
+        if comment is not None:
+            data[name]['comment'] = comment
+        if active:
+            data[name]['active'] = True
+        else:
+            data[name]['active'] = False
+        self._write(preset_type, data)
+
+    def sync(self):
+        """
+        Synchronize the presets with the database.
+        """
+        logger.debug('call %s presets.sync()', self._device.name)
+        self._remove_methods()
+        self._cache = {}
+        for preset_type in self._paths.keys():
+            try:
+                self._cache[preset_type] = self._read(preset_type)
+            except FileNotFoundError:
+                pass
```

Comment:
Might be nice to put a `debug` message here. Catching exceptions with only a pass makes me nervous, even if we except that not every device has every preset type.

Reply:
I understand. The caveat here is that this will be called for every device that doesn't have a preset defined.